### PR TITLE
Fixing of Linux support and incorrect flag detection.

### PR DESCRIPTION
**Pull request by linuxnoodle**
Bugs fixed:
    Starting at 0 for the index of program instance.
    Getting arguments through argv[] was out of sync with actual indices.
Detects:
    Non-existent files being run.
    Non-integers being supplied to count.

### CODE REVIEW

File: Main.cpp [hunk 1]
```diff
@@ -30,69 +28,85 @@ double compute_average(std::vector<int>& vi) {
...
-#elif defined(linux)
-	cout << "You are running on a linux system. Some things may be broken. as this is a native app to windows.\n";
-#elif defined(__unix__)
-	cout << "You are running on a unix system. Some things may be broken. as this is a native app to windows.\n";
-#else
-	cout << "You are not running windows. Some things may be broken. as this is a native app to windows.\n";
-#endif
+	// if the user is on anything but windows tell them it might not work
+	#ifdef _WIN32
+		HANDLE hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
```
Comment (Waves-rgb) [inline Main.cpp:33]: Broken.

File: Main.cpp [hunk 2]
```diff
@@ -30,69 +28,85 @@ double compute_average(std::vector<int>& vi) {
...
 	// set the file to the last argument
-	const char* file = argv[sizeof(argv) + 1];
-	cout << "Running " << file << " " << count << " times. and writing output to ";
-	cout << output << "\n";
+	if (argv[argc - 1][0] == '.' || argv[argc - 1][0] == '/'){
+		strcpy(file, argv[argc - 1]);
+	} else {
+		strcpy(file, "./");
+		strcat(file, argv[argc - 1]);
+	}
```
Comment (Waves-rgb) [inline Main.cpp:92]: Breaks the program on windows.

File: Main.cpp (hunk 1 above)
Comment (Waves-rgb) [inline Main.cpp:33]: Program will not compile without a matching #endif

**Review by Waves-rgb — CHANGES_REQUESTED** [review]
I have added comments for things that are broken. please fix them.

File: Main.cpp (hunk 1 above)
Comment (linuxnoodle) [reply Main.cpp:33]: Shoot, I accidently removed the #ENDIF. Guess 2 AM does this stuff.

**Pull request merged**